src/graphic/minimap_renderer.cc

```diff
@@ -256,3 +261,19 @@ std::unique_ptr<Texture> draw_minimap(const EditorGameBase& egbase,
 
 	return texture;
 }
+
+int scale_map(const Widelands::Map& map, const bool zoom) {
+	// If a map is wider than 300px we don't scale. Otherwise we fit as much as possible into a 300px
```

Comment:
Is map.get_width() the map width in pixels or in triangles? Or is this the same?
Lots of magic numbers here. Where do they come from? My guess: A result of trial-until-it-looks-nice.

Reply:
It is the map width in pixel. `map.get_width()` on a 400×400 map would give you 400.

The numbers are easily explained.

The max width for the minimap is 600px. This is because the 6 buttons used in the minimap have a width 100 each. That is the maximum we can scale a map to. if we have more than 600 pix the minimap layout completely breaks, as it does ATM.

Without zoom, we try to fit the map in a 300px wide window. And for maps with more than 150 in a 400px wide window. This makes sure that we don't end up with a minimap window that is too small (e.g. 64px or 151px).